packages/common/src/converter/data-converter.ts

```diff
@@ -182,8 +188,21 @@ export function mapToPayloadsSync<K extends string>(
   ) as Record<K, Payload>;
 }
 
-export const defaultDataConverter = new CompositeDataConverter(
-  new UndefinedPayloadConverter(),
-  new BinaryPayloadConverter(),
-  new JsonPayloadConverter()
-);
+export interface DefaultDataConverterOptions {
+  protobufClasses?: Record<string, unknown>;
+}
+
+export class DefaultDataConverter extends CompositeDataConverter {
+  constructor({ protobufClasses }: DefaultDataConverterOptions = {}) {
+    // Match the order used in other SDKs
+    // Java: https://github.com/temporalio/sdk-java/blob/0da3149c8eb16454c30fb7bda32d28f2539f3f3c/temporal-sdk/src/main/java/io/temporal/common/converter/DefaultDataConverter.java#L44
+    super(
+      new UndefinedPayloadConverter(),
+      new BinaryPayloadConverter(),
+      new ProtobufPayloadConverter(protobufClasses),
```

Comment:
Note, there are actually two types of payload encoding: JSON and binary. Unfortunately it appears the Java default does not include the binary one (not sure why), and unfortunately this one does not include the JSON one.

Maybe the Go SDK at https://github.com/temporalio/sdk-go/blob/5e5645f0c550dcf717c095ae32c76a7087d2e985/converter/default_data_converter.go#L28 is better to look at here because it is more robust.

Cross SDK compatibility with these is important. I will followup on the Java SDK to see why the `binary/protobuf` is not supported by default. In the meantime, I think we need both here and with JSON being listed first.

Reply:
Ah, thanks! Will update.

Reply:
+1 to what @cretz said